subsystem/hcp/default_ingress_resource_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcp
+
+import (
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"                      // nolint
+	. "github.com/onsi/gomega"                         // nolint
+	. "github.com/onsi/gomega/ghttp"                   // nolint
+	. "github.com/openshift-online/ocm-sdk-go/testing" // nolint
+)
+
+var _ = Describe("rhcs_cluster_rosa_hcp - default ingress", func() {
+
+	defaultDay1Template := `{
```

Comment:
One thing I thought about doing is using the cmv1.IngressBuilder and cmv1.MarshallIngress instead of supplying the whole string for the jsons, what do you think? [Here's](https://github.com/gdbranco/terraform-provider-ocm/blob/67217d584cffde8425318a606b8913e679d064e0/subsystem/hcp/cluster_create_test.go#L61) one a did for cluster

Reply:
Also, just noticed this is a page. cmv1 doesn't have a marshall option for pages, so let's keep it this way

Reply:
for the clusterready and single ingress, I can generate the json using that way. But for the ingresslist, seems the ingresslistbuilder and marshalIngressList cannot build out the required json.
So still leave the whole string for the ingresslist, aka `defaultDay1Template`, and I will change others. Are you ok with this? @gdbranco